libs/sdk-core/src/persist/transactions.rs

```diff
@@ -196,8 +238,22 @@ impl SqliteStorage {
             .as_str(),
         )?;
 
+        let mut params: HashMap::<String, String> = HashMap::new();
+
+        if let Some(metadata_filters) = &req.metadata_filters {
+            metadata_filters.iter().enumerate().for_each(|(i, PaymentMetadataFilter { search_path, search_value })| {
+                params.insert(format!(":search_path_{i}"), format!("'$.{search_path}'"));
+                params.insert(format!(":search_value_{i}"), format!("'{search_value}'"));
+            })
+        }
```

Comment:
Based on the solution proposed by @roeierez and @dangeross , I decided to opt for their approach. This still does not work, but at the same time the parser does not complain of any parsing errors. I am thinking the issue is now related to the JSON Path, as maybe it is not evaluated as such due to the fact that the `$` is part of the prepared statement, perhaps? I have verified the values get indeed passed to `query_map` as intended, therefore I imagine it to be a SQL-related issue.

Reply:
The test you added did pass using the commit I posted: https://github.com/breez/breez-sdk/commit/7a324252a22e2954ad93cb83e55890b1c836cdbc

Reply:
@roeierez Thanks for the swift reply, turns out the quotes were not needed at all. Made it work with the HashMap, pushing rn!